### PR TITLE
Feat/chain switcher implemented

### PR DESCRIPTION
**Pull request by Nainish-Rai**
# Pull Request Template

## Description

Check whether the user have connected to base sepolia or not, if not then a popup will appear for user to change the chain

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so others can reproduce. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Screenshots (if applicable)
image

### CODE REVIEW

File: components/chain-switcher.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { Card, CardContent, CardTitle } from "@/components/ui/card";
+import { Button } from "@/components/ui/button";
+import Link from "next/link";
+import { useSwitchActiveWalletChain } from "thirdweb/react";
+import { baseSepolia, sepolia } from "thirdweb/chains";
+
+const ChainSwitcher = ({
+  setActiveChainChanged,
+}: {
+  setActiveChainChanged: any;
```
Comment (arindam-monet) [inline components/chain-switcher.tsx]: change the type to () => boolean instead of any.

File: components/customer-dashboard.tsx [hunk 2]
```diff
@@ -26,6 +26,10 @@ const CustomerDashboard = () => {
     null
   );
   const [onChainPoints, setOnChainPoints] = useState(0);
+  const [dataWithOnChainPoints, setDataWithOnChainPoints] = useState<
+    any | null
```
Comment (arindam-monet) [inline components/customer-dashboard.tsx]: can we avoid using the any type unless there is a valid reason?

File: context/auth-context.tsx [hunk 3]
```diff
@@ -163,9 +187,12 @@ export const AuthProvider = ({ children }: { children: ReactNode }) => {
     isProtectedRoute,
     authResponse,
     activeWalletChanged,
+    activeChainChanged,
+    setActiveChainChanged,
   } = useAuthProvider();
 
   const userStore = useUserStore();
+  const switchChain = useSwitchActiveWalletChain();
```
Comment (arindam-monet) [inline context/auth-context.tsx]: this seems to be duplicate. This statement appears in line 57 as well.
Also this this hook seems unused in auth-context.tsx.

**Review by arindam-monet — CHANGES_REQUESTED** [review]
please resolve the PR comments

File: components/customer-dashboard.tsx [hunk 4]
```diff
@@ -64,8 +68,34 @@ const CustomerDashboard = () => {
     enabled: !!customerStore?.customer?.id,
   });
 
+  // console.log(customerPointsResponse, "customerPointsResponse");
   const { mutate: sendTransaction, isPending, isError } = useSendTransaction();
 
+  const fetchOnChainPointsForAllTokens = async () => {
```
Comment (arindam-monet) [inline components/customer-dashboard.tsx]: I believe there is another PR for fetching the customer's on-chain points. Then why is this part of this PR as well?
You can probably merge develop into this branch.

File: app/(pages)/marketplace/page.tsx [hunk 5]
```diff
@@ -169,7 +169,11 @@ const MarketplacePage = () => {
         <div className="mt-4">
           <DataTable
             columns={PointsListColumns}
-            data={blockchainData}
+            data={
```
Comment (arindam-monet) [inline app/(pages)/marketplace/page.tsx]: why this has changes related to marketplace?

File: context/auth-context.tsx [hunk 6]
```diff
@@ -114,14 +120,29 @@ const useAuthProvider = () => {
     }
   }, [authError]);
 
+  useEffect(() => {
+    if (!chainId) return;
+    // console.log(chainId, baseSepolia, chainId.id === baseSepolia.id);
```
Comment (anujkmonetwork) [inline context/auth-context.tsx]: Remove this log.

File: context/auth-context.tsx [hunk 7]
```diff
@@ -114,14 +120,29 @@ const useAuthProvider = () => {
...
+
   useEffect(() => {
     if (!activeAccount) return;
     if (!userWallerAddress) {
       setUserWalletAddress(activeAccount.address);
       return;
     }
     if (activeAccount.address !== userWallerAddress) {
-      console.log(activeAccount.address, 'active', userWallerAddress, 'user address')
+      console.log(
```
Comment (anujkmonetwork) [inline context/auth-context.tsx]: Remove this log.

File: components/chain-switcher.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+}) => {
+  const switchChain = useSwitchActiveWalletChain();
+  return (
+    <div className="bg-background/80 w-full fixed h-full backdrop-blur-sm flex justify-center items-center">
+      <Card className="backdrop-blur-sm bg-muted-foreground/20 relative">
+        <div className="p-4">
+          <div className="flex flex-col">
+            <h1 className="text-2xl font-bold">Different Chain Detected!</h1>
+            <p className="text-sm mt-2 max-w-sm text-muted-foreground">
+              We have detected you are using a different chain. Please switch to
```
Comment (anujkmonetwork) [inline components/chain-switcher.tsx]: You’re on a different network. Switch to **BaseSepolia** to keep going. 🚀
Change text to this

File: components/chain-switcher.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+              <div className="mt-8 flex flex-col justify-center gap-4">
+                <Button
+                  onClick={() =>
+                    switchChain(baseSepolia).then(() =>
+                      setActiveChainChanged(true)
+                    )
+                  }
+                  className="bg-blue-600 px-12 py-3 text-sm font-medium text-white hover:bg-purple-600 hover:text-white focus:outline-none focus:ring active:text-opacity-75 sm:w-full"
+                >
+                  Switch to Base Sepolia
```
Comment (anujkmonetwork) [inline components/chain-switcher.tsx]: Connect to **Base Sepolia!** 🌟
Change text to this.

**Review by anujkmonetwork — CHANGES_REQUESTED** [review]

**Pull request merged**